BE/Tests/PRMergescore.spec.ts

```diff
@@ -131,6 +131,6 @@ describe("MergeRestriction_Scorer with real repository", () => {
         //the response you get wherever you query for this should be mocked, if the repo changes, the test expected value changes and fails.
         const score = await MergeRestriction_Scorer(repository);
         //console.log(`Merge Restriction Score for cloudinary/cloudinary_npm: ${score}`);
-        expect(score).toBe(0); // Adjust this expectation based on real data
-    });
+        expect(score).toBe(0.69); // Adjust this expectation based on real data
```

Comment:
This is okay, but consider that any time the data about the package from GQL changes. The response value will also change, breaking the test. The test would fail for something that isn't exactly a failure. 

If we mock fetch we can control the return. This will isolate the test and allow you to test only that your functionality works. 

Optional, but this test could fail after submission while it works properly now.

Reply:
```ts
export const getFetchSpy = <T>(returnBody: T, statusCode: number, throwAnError?: boolean) =>
    jest
        .spyOn(global, 'fetch')
        .mockImplementation(async (input: string | URL | Request, init?: RequestInit | undefined) => {
            if (throwAnError === true) {
                throw new Error('fetch spy threw an error');
            }
            //A blob
            const blob = new Blob([JSON.stringify(returnBody, null, 2)], {
                type: 'application/json',
            });
            //the mock response object fetch will return when called
            return new Response(blob, { status: statusCode });
        });
```

This will allow you to set the return body for your fetch calls during MergeRestriction_Scorer. Could be useful for everyone if you do end up mocking.